zeebe/engine/src/main/java/io/camunda/zeebe/engine/metrics/JobProcessingMetrics.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.metrics;
+
+import io.camunda.zeebe.engine.metrics.EngineMetricsDoc.EngineKeyNames;
+import io.camunda.zeebe.engine.metrics.EngineMetricsDoc.JobAction;
+import io.camunda.zeebe.util.collection.Table;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import net.jcip.annotations.NotThreadSafe;
+
+/**
+ * This class is not thread safe, and is meant to be used only in the context of the engine actor.
+ *
+ * <p>If you need to use it from the scheduling actor, or from somewhere else, then you need to
+ * first make this thread safe.
+ */
+@NotThreadSafe
+public final class JobProcessingMetrics {
```

Comment:
:thought_balloon: :crossed_fingers: renaming this class will not create more conflicts across all backports

Reply:
I renamed it to differentiate it from the others, since its accessed during processing and not from the state. We've discussed with Chris in the past to extract the state classes into their own modules, and the fact they're opened outside of the engine module is a strong hint to that. Otherwise you couple things to the engine (and its processing classes) only for the state, leading to a "wide" module.